internal/pkg/k8s/check.go

```diff
@@ -0,0 +1,17 @@
+package k8s
```

Comment:
Can we maybe rename this to k8sutil or something? Only k8s feels a little off. Not a blocker.

Reply:
not a big fan of "util" in package names but also had the same feeling. essentially this package is for all things k8s that didnt fit anywhere else hence the name :|

Reply:
sorted by natural refactor based on prior comments :)

Reply:
Nice! :)